ingest/ingest_pipeline.py

```diff
@@ -293,6 +299,9 @@ def upload_metadata_to_bq(self):
                 return 1
         return 0
 
+    @custom_metric(
+        "ingest-pipeline:expression:ingest", settings.get_study, settings.get_study_file
```

Comment:
This approach lets us analyze across jobs -- e.g. median duration of the `ingest-pipeline:expression:ingest` event across many ingests.  The downside is that it's quite network-intensive (one log request per function call) given that the only props that change are `functionName` (which maps 1-to-1 with `event`) and `perfTime`.

A complementary approach to consider: we could also accumulate these events and their `perfTimes` in your new singleton object.  Then, when ingest completes (successfully or not), we could log _one event_ with an ordered list of `perfTime:<event_order>:<event_name>` props for all events in a given ingest.  

This would follow a pattern we use in `single_cell_portal_core` (see [diagram and explanation](https://github.com/broadinstitute/single_cell_portal_core/pull/716)) that we've found helpful in Mixpanel triage.  I don't think it's blocking or necessarily in scope here -- just thought I'd note it for your consideration.  It would augment your existing work here.

Reply:
Ohhhhh ok. I'll make this update in the other PR. I appreciate your attention to detail on this!